src/coreclr/ilasm/GrammarExtractor/Program.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Text.RegularExpressions;
+using static Patterns;
+
+if (args.Length != 1)
+{
+    Console.Error.WriteLine("Usage: <file>");
+    return 1;
+}
+
+string filePath = args[0];
+
+string fileContent = File.ReadAllText(filePath);
+
+var match = GetRegexExtractMarkers().Match(fileContent);
+if (!match.Success)
+{
+    Console.Error.WriteLine("Could not find %% markers");
+    return 1;
+}
+
+//string prefix = match.Groups[1].Value;
+string grammar = match.Groups[2].Value;
+
+// Remove any text in {}
+var regexRemoveTextInBraces = GetRegexRemoveTextInBraces();
+string previousGrammar;
+
+do
+{
+    previousGrammar = grammar;
+    grammar = regexRemoveTextInBraces.Replace(grammar, "$1");
+} while (grammar != previousGrammar);
+
+// Change keyword identifiers into the string they represent (lowercase)
+grammar = GetRegexKeywordIdentifiers().Replace(grammar, m => $"'{m.Groups[1].Value.ToLowerInvariant()}'");
+
+// Change assembler directives into their string (lowercase with a period)
+grammar = GetRegexAssemblerDirectives().Replace(grammar, m => $"'.{m.Groups[1].Value.ToLowerInvariant()}'");
+
+// Handle special punctuation
+grammar = GetRegexEllipsis().Replace(grammar, "'...'");
+grammar = GetRegexDcolon().Replace(grammar, "'::'");
+
+// Remove TODO comments
+grammar = GetRegexRemoveTodoComments().Replace(grammar, "\n");
+
```

Comment:
```suggestion
```
Is this needed?

Reply:
We don't have any TODOs left. Removed the regex pattern and usage.